File: clientLeger/app/src/main/java/com/projet/clientleger/data/api/model/chat/ChatRoomHistory.kt
package com.projet.clientleger.data.api.model.chat

import com.projet.clientleger.data.model.chat.MessageChat
import kotlinx.serialization.Serializable

@Serializable
data class ChatRoomHistory(val roomName: String, val messages: ArrayList<RoomMessage>){
    fun messagesToMessagesChat():ArrayList<MessageChat>{
        val list = ArrayList<MessageChat>()

        for(message in messages){
            list.add(message.toMessageChat())
        }
        return list
    }
}

File: common/communication/chat-room-history.ts
import { ChatMessage, SystemMessage } from "./chat-message";

export interface ChatRoomHistory {
    roomName: string;
    messages: ChatRoomMessage[];
}

export interface ChatRoomMessage extends ChatMessage {
    senderAccountId: string;
    roomName: string;
}

export interface ChatRoomSystemMessage extends SystemMessage {
    roomName: string;
}